photon-core/src/main/java/org/photonvision/vision/camera/USBCameraSource.java

```diff
@@ -60,67 +66,173 @@ public USBCameraSource(CameraConfiguration config) {
+            retProp = null;
+        }
+
+        return Optional.ofNullable(retProp);
     }
 
-    void disableAutoFocus() {
-        if (getCameraConfiguration().cameraQuirks.hasQuirk(CameraQuirk.AdjustableFocus)) {
+    /**
+     * Forgiving "set this property" action. Produces a debug message but skips properties if they
+     * aren't supported. Errors if the property exists but the set fails.
+     *
+     * @param property
+     * @param value
+     */
+    private void softSet(String property, int value) {
+        VideoProperty prop = camera.getProperty(property);
+        if (prop.getKind() == VideoProperty.Kind.kNone) {
+            logger.debug("No property " + property + " for " + camera.getName() + " , skipping.");
+        } else {
             try {
-                camera.getProperty("focus_auto").set(0);
-                camera.getProperty("focus_absolute").set(0); // Focus into infinity
+                prop.set(value);
             } catch (VideoException e) {
-                logger.error("Unable to disable autofocus!", e);
+                logger.error("Failed to set " + property + " for " + camera.getName() + " !", e);
             }
         }
     }
 
+    private void printCameraProperaties() {
+        VideoProperty[] cameraProperties = null;
+        try {
+            cameraProperties = camera.enumerateProperties();
+        } catch (VideoException e) {
+            logger.error("Failed to list camera properties!", e);
+            return;
+        }
+
+        if (cameraProperties != null) {
+            String cameraPropertiesStr = "Cam Properties Dump:\n";
+            for (int i = 0; i < cameraProperties.length; i++) {
+                cameraPropertiesStr +=
+                        "Name: "
+                                + cameraProperties[i].getName()
+                                + ", Kind: "
+                                + cameraProperties[i].getKind()
+                                + ", Value: "
+                                + cameraProperties[i].getKind().getValue()
+                                + ", Min: "
+                                + cameraProperties[i].getMin()
+                                + ", Max: "
+                                + cameraProperties[i].getMax()
+                                + ", Dflt: "
+                                + cameraProperties[i].getDefault()
+                                + ", Step: "
+                                + cameraProperties[i].getStep()
+                                + "\n";
+            }
+            logger.debug(cameraPropertiesStr);
+        }
+    }
+
+    private void setAllCamDefaults() {
+        // Common settings for all cameras to attempt to get their image
+        // as close as possible to what we want for image processing
+        softSet("image_stabilization", 0); // No image stabilization, as this will throw off odometry
+        softSet("power_line_frequency", 2); // Assume 60Hz USA
```

Comment:
```suggestion
        softSet("power_line_frequency", 2); // Assume 60Hz USA 🇺🇸
```
til java supports emojis

Reply:
![image](https://github.com/user-attachments/assets/6cebd247-ac07-4890-84bc-d44b46d22455)
 :( vscode keeps not showing it